service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -142,12 +152,57 @@ private boolean registerSchemas() {
       boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
       LOGGER.info("schemaId {} exists {}", schemaId, exists);
       if (!exists) {
-        if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
-          return false;
+        if (serviceCenterEnv()) {
+          if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
+            return false;
+          }
+        }
+      } else {
+        //查询是否存在schema,如果不存在则注册
+        String serviceId = microservice.getServiceId();
+        String schema = srClient.getSchema(serviceId, schemaId);
+        if (null == schema) {
+          LOGGER.warn("schemacontent {}/{} not exist.", serviceId, schemaId);
+          if (!srClient.registerSchema(serviceId, schemaId, content)) {
+            return false;
+          }
+        } else {
+          //查询是否存在摘要,不存在则注册
+          String schemaSummary = srClient.getSchemaSummary(serviceId, schemaId);
+          if (null == schemaSummary) {
+            LOGGER.warn("schemasummary {}/{} not exist.", serviceId, schemaId);
+            if (!srClient.registerSchema(serviceId, schemaId, content)) {
+              return false;
+            }
+          } else {
+            //比较本地摘要,如果不匹配,查询SC环境
+            String localSummary = Sha.getSHA256StrJava(content);
+            if (!localSummary.equals(schemaSummary)) {
+              LOGGER.warn("localsummary is not equal to what is already exist in servicecenter");
+              if (serviceCenterEnv()) {
+                if (!srClient.registerSchema(serviceId, schemaId, content)) {
+                  return false;
+                }
+              }
+            }
+          }
         }
       }
     }
 
     return true;
   }
+
+  private boolean serviceCenterEnv() {
```

Comment:
serviceCenterEnv is not clear
it seems that to determine if it is dev mode.

Reply:
Use String instead of boolean to do the return value type, and then judge?

Reply:
only in the dev can perform changes currently, so determine if it is dev mode in the method,we can only care about it'll return true or false

Reply:
yes.
but method name is serviceCenterEnv, not relative to dev mode.

Reply:
ok,i've changed the metnod's name to serviceCenterEnvIsDev